modules/javafx.graphics/src/main/native-glass/win/GlassClipboard.cpp

```diff
@@ -1302,7 +1302,10 @@ JNIEXPORT jobjectArray JNICALL Java_com_sun_glass_ui_win_WinSystemClipboard_popM
                         //LPFILEGROUPDESCRIPTORW for MS_FILE_DESCRIPTOR_UNICODE
                         //LPFILEGROUPDESCRIPTORA for MS_FILE_DESCRIPTOR
                         LPFILEGROUPDESCRIPTORW pdata = reinterpret_cast<LPFILEGROUPDESCRIPTORW>(me.getMem());
-                        if (pdata->cItems > 0) {
+                        jsize bufferSize = me.size() - sizeof(UINT);
+                        if ((pdata->cItems > 0) &&
+                            (bufferSize / pdata->cItems >= itemSize))
```

Comment:
Instead of discarding all the data, have you considered reading `min(pdata->cItems, bufferSize / itemSize)` items?

Reply:
I thought about it, but since failing this test means that `cItems` is invalid, there is no reason to believe that the data that follows it is any less invalid.

Reply:
Then shouldn't we also not trust the data if `bufferSize` is larger than it needs to be? The documentation of `FILEGROUPDESCRIPTORA/W` says that `cItems` should correspond exactly to the numer of items in the array that follows.

Reply:
> Then shouldn't we also not trust the data if bufferSize is larger than it needs to be?

Yes, that's a good point.

In order to avoid integer overflow, I'll probably leave the two existing tests, and add a third (or else do the test using a `jlong`). Something like this should work:

```
                        jsize bufferSize = me.size() - sizeof(UINT);
                        if ((pdata->cItems > 0) &&
                            (bufferSize / pdata->cItems >= itemSize) &&
                            (bufferSize == pdata->cItems * itemSize))
```

I'll update this tomorrow.

Reply:
Fixed. I decided it was cleaner to use `jlong` and just have the equality check.